controllers/complianceeventsapi/server.go

```diff
@@ -309,6 +329,30 @@ func (s *ComplianceAPIServer) Start(ctx context.Context, serverContext *Complian
 	}
 }
 
+func getUserKubeConfig(config *rest.Config, w http.ResponseWriter, r *http.Request) (*rest.Config, error) {
+	userConfig := &rest.Config{
+		Host:    config.Host,
+		APIPath: config.APIPath,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAFile:     config.TLSClientConfig.CAFile,
+			CAData:     config.TLSClientConfig.CAData,
+			ServerName: config.TLSClientConfig.ServerName,
+			// For testing
+			Insecure: config.TLSClientConfig.Insecure,
+		},
+	}
+
+	userConfig.BearerToken = parseToken(r)
+
+	if userConfig.BearerToken == "" {
+		writeErrMsgJSON(w, "The Authorization header is not set", http.StatusForbidden)
```

Comment:
Could you let the calling function do this instead? That would allow you to use this function in `canRecordComplianceEvent`.